src/sources/kubernetes_logs/group_watcher.rs

```diff
@@ -0,0 +1,80 @@
+use std::collections::HashMap;
+use std::time::{Duration, Instant};
+
+#[derive(Clone, Debug)]
+pub struct FileCounter {
+    start_time: Instant,
+    lines_read: usize,
```

Comment:
Consider setting limit in bytes/sec if possible - I understand that may be more complex. The limits are to protect network bandwidth or disk storage, both measured in bytes. There's no easy way to predict the average byte size of a line.